### ISSUE TITLE
Supervisorctl "stop all" not parallel

### ISSUE THREAD

**Issue by TomWis97** [ansible-collections/community.general#8159]
### Summary

When running `supervisorctl stop all` on the CLI, processes are stopped parallel. However, when using the `community.general.supervisorctl` module with `name=all,state=stopped`, processes are being stopped sequentially. When every process takes a few seconds to stop (and you have a few of these), this will a while.

### Issue Type

Bug Report

### Component Name

supervisorctl

### Ansible Version

```console (paste below)
$ ansible --version
ansible [core 2.16.5]
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.10/dist-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] (/usr/bin/python3)
  jinja version = 3.1.3
  libyaml = True
```

### Community.general Version

```console (paste below)
$ ansible-galaxy collection list community.general
Collection        Version
----------------- -------
community.general 8.5.0
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```

### OS / Environment

Not environment dependant. But I reproduced on the `ubuntu:latest` image.

### Steps to Reproduce

1. Create a script which takes a few seconds to respond to a SIGTERM in `/sigterm.sh`:
```bash
#!/usr/bin/bash
_term() {
    echo "Caught SIGTERM!"
    sleep 2
    echo "Exiting..."
    exit 0
}

trap _term SIGTERM

while true
do
    sleep 1
done
```
2. Make script executable with `chmod +x /sigterm.sh`.
3. Build supervisord config file with 20 apps with this command:
```bash
for x in {1..20}; do cat << EOF >> /etc/supervisor/conf.d/test.conf; done
[program:test-app-$x]
command=/sigterm.sh
autostart=true
autorestart=true
stopwaitsecs=5

EOF
```
4. Start supervisord. This will start the 20 programs.
5. Run `supervisorctl stop all` while looking at supervisord's logs. Observer that all 20 programs are being stopped in parallel. This command returns within a few seconds.
6. Start the programs again with `supervisorctl start all`.
7. Stop the apps through Ansible with an ad-hoc command: `ansible -m community.general.supervisorctl -a 'name=all state=stopped' localhost`. 

### Expected Results

The ad-hoc finishes within 5 seconds.

### Actual Results

The ad-hoc command takes more than a minute.

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibullbot** [comment]
Files identified in the description:
* [`plugins/modules/supervisorctl.py`](https://github.com/ansible-collections/community.general/blob/main/plugins/modules/supervisorctl.py)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/devel/ISSUE_HELP.md)

**Comment by ansibullbot** [comment]
cc @inetfuture @mattupstate
[click here for bot help](https://github.com/ansible/ansibullbot/blob/devel/ISSUE_HELP.md)

**Comment by felixfontein** [comment]
I don't think this is a bug, but a feature request. The module does not claim that it does exactly the same as a similar command line, just that it makes sure that all processes are stopped (in your case).

**Comment by TomWis97** [comment]
Ah, I understand. Would it be possible to convert this bug report to a feature request, or should I create a new issue?
